### DISCUSSION TITLE
Incorrect statement section?

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#96] [General]
Hi Bruce,
We today noticed an error processing an example script, the single statement looks like it has been sectioned into multiple statements instead of just one.

We are using version 2.1.0.

Example input script:
`SELECT CAST(123.45 AS INT), CAST(456.321 AS VARCHAR(10)), Column_1 FROM MyTable`

Output:
`Statement 1:  SELECT CAST ( 123.45 AS INT ) , CAST ( 456.321 AS VARCHAR 10
        Token type: Keyword     Text: SELECT
        Token type: SystemIdentifier    Text: CAST
        Token type: Character   Text: (
        Token type: NumericLiteral      Text: 123.45
        Token type: Keyword     Text: AS
        Token type: Identifier  Text: INT
        Token type: Character   Text: )
        Token type: Character   Text: ,
        Token type: SystemIdentifier    Text: CAST
        Token type: Character   Text: (
        Token type: NumericLiteral      Text: 456.321
        Token type: Keyword     Text: AS
        Token type: Identifier  Text: VARCHAR
        Token type: NumericLiteral      Text: 10

Statement 2:  )
        Token type: Character   Text: )

Statement 3:  ) , Column_1 FROM MyTable
        Token type: Character   Text: )
        Token type: Character   Text: ,
        Token type: Identifier  Text: Column_1
        Token type: Keyword     Text: FROM
        Token type: Identifier  Text: MyTable`

It looks like the parser has broken this single select into 3 statements and this caused execution part errored out as we execute based on the statements have been sectioned. This version 2.1.0 used to be working fine after fixing the missing "AS" in CAST statement in ver 2.0.1. It may be like that for some time already, but we only found out today.

I also tried ver 2.2.0, 2.2.1, the results are the same as above.

Could you please check if anything that may be causing this?

Thank you very much Bruce.

Regards
Paul

**Comment by bruce-dunwiddie** [comment]
Ok. Thanks for letting me know. I'll confirm tomorrow and then dig into it.

**Reply by cptb** [reply]
Thanks for your quick response. I guess it may be the bracket that breaks the logic. E.g.  _AS VARCHAR(8))_

**Reply by bruce-dunwiddie** [reply]
Verified. And issue created, https://github.com/bruce-dunwiddie/tsql-parser/issues/98 .

Yes, it appears to be related to getting out of whack with the parens. Hopefully it's an easy solve and I can put out a fix within a few days. I'm not currently expecting it to be too nasty.

**Reply by bruce-dunwiddie** [reply]
Ok, CAST function parsing should now be fixed in https://www.nuget.org/packages/TSQL.Parser/2.2.2 .

**Reply by cptb** [reply]
Thank you very much, Bruce. The fix works great. You are a legend.
